pkg/services/correlations/database.go

```diff
@@ -77,6 +77,61 @@ func (s CorrelationsService) deleteCorrelation(ctx context.Context, cmd DeleteCo
 	})
 }
 
+func (s CorrelationsService) updateCorrelation(ctx context.Context, cmd UpdateCorrelationCommand) (Correlation, error) {
+	correlation := Correlation{
+		UID:       cmd.UID,
+		SourceUID: cmd.SourceUID,
+	}
+
+	err := s.SQLStore.WithTransactionalDbSession(ctx, func(session *sqlstore.DBSession) error {
+		query := &datasources.GetDataSourceQuery{
+			OrgId: cmd.OrgId,
+			Uid:   cmd.SourceUID,
+		}
+		if err := s.DataSourceService.GetDataSource(ctx, query); err != nil {
+			return ErrSourceDataSourceDoesNotExists
+		}
+
+		if query.Result.ReadOnly {
+			return ErrSourceDataSourceReadOnly
+		}
+
+		if cmd.Label == nil && cmd.Description == nil {
+			return ErrUpdateCorrelationEmptyParams
+		}
+		update := Correlation{}
+		if cmd.Label != nil {
+			update.Label = *cmd.Label
+			session.MustCols("label")
```

Comment:
Nit.

Reply:
I'd have to implement a `getCorrelation` function. given it's coming in a follow-up PR (https://github.com/grafana/grafana/pull/52517) i'd rather leave this as it is and change it later if necessary.

Reply:
Do you want to allow user do a patch update of correlations on org_id? if yes, you might want to do it now, since org_id = 0 would be ignored by default with update function in xorm. It might be a bug if so, otherwise it could be fine.

Reply:
There's no org_id field on correlations, the only patchable attributes are label and description.
The only use for org_id here is to check (via passing it to getDatasource) if the user has access to the given source datasource. Or maybe I'm missing something?

 i believe getDatasource returns an error in case orgId is 0 and the update would fail.

I'll double check this.

Reply:
seems to work as expected 👍